calico_kubernetes/calico_kubernetes.py

```diff
@@ -265,28 +265,58 @@ def _assign_container_ip(self):
 
         :return IPAddress which has been assigned
         """
+        def _assign(ip):
```

Comment:
Cool, not a massive issue so I won't block this going in if you can't come up with something cleaner.